### ISSUE TITLE
Unable to start immich after upgrade (v1.102.3 -> v1.107.2) and funky/bad pgvector state due to `face_search` not yet being created

### ISSUE THREAD

**Issue by lgo** [immich-app/immich#10871]
### The bug

I just got around to upgrading things and immich didn't want to come up right away. I didn't keep a close track of what I did which I'm regretting now as I might have broke another thing in the process that made this a bit more complicated[0]. This was my crash log:
```
Detected CPU Cores: 40
Starting api worker
Starting microservices worker
[Nest] 7  - 07/05/2024, 4:02:58 AM     LOG [Microservices:EventRepository] Initialized websocket server
[Nest] 17  - 07/05/2024, 4:02:58 AM     LOG [Api:EventRepository] Initialized websocket server
[Nest] 7  - 07/05/2024, 4:03:01 AM    WARN [Microservices:DatabaseRepository] Could not reindex index face_index. Attempting to auto-fix.
[Nest] 7  - 07/05/2024, 4:03:02 AM    WARN [Microservices:DatabaseService] Could not run vector reindexing checks. If the extension was updated, please restart the Postgres instance.
TypeError: Cannot read properties of undefined (reading 'dimsize')
    at DatabaseRepository.getDimSize (/usr/src/app/dist/repositories/database.repository.js:162:31)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async DatabaseRepository.reindex (/usr/src/app/dist/repositories/database.repository.js:96:33)
    at async /usr/src/app/dist/services/database.service.js:126:21
    at async /usr/src/app/dist/repositories/database.repository.js:177:23
microservices worker exited with code 1
```

Looking at the code, there's this re-indexing step (potentially due to my broken pgvector state / weirdness). https://github.com/immich-app/immich/blob/e32ce82179b54be65833aa12a71c21493880f53f/server/src/repositories/database.repository.ts#L87-L88

but the `REINDEX INDEX` is failing, hitting the auto-fix, then failing and crashing on `getDimSize`. https://github.com/immich-app/immich/blob/6030349a6fffe9f66dd78ccb4d05c5d38c362d3e/server/src/repositories/database.repository.ts#L102

importantly, I didn't have the `face_search` table which was added in `v1.107.0` so this pre-migration step is always going to fail on `getDimSize`.

I was able to upgrade to `v1.106.4`, have the auto-fix complete, and then successfully upgrade past `v1.107.0`.

---
[0]: I thought my database was probably in an OK state for `pgvector` as it was already upgraded to 0.2.0 (specifically `tensorchord/pgvecto-rs:pg16-v0.2.0`).

I was getting some errors where I mindlessly ran some upgrade Postgres commands related to the 0.1.11 to 0.2.0 upgrade (manually set the version, attempt upgrade, etc.). I ended up running a few more things to get immich to make a bit more progress on boot:
```sql
-- in immich database
UPDATE pg_catalog.pg_extension SET extversion = '0.2.0' WHERE extname = 'vectors';
-- in `postgres` database. this was failing for a bit
GRANT SELECT ON TABLE pg_vector_index_stat to <immuch_user>
```

Everything else in the report is after having broken pgvector then unwedge it with that.

### The OS that Immich Server is running on

Debian

### Version of Immich Server

v1.107.2

### Version of Immich Mobile App

n/a

### Platform with the issue

- [X] Server
- [ ] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
services:
  {{ service_name }}:
    container_name: {{ service_name }}
    image: ghcr.io/immich-app/immich-server:{{ service_version }}
    volumes:
      - {{ document_root }}/immich:/usr/src/app/upload
      - /etc/localtime:/etc/localtime:ro
    environment:
      IMMICH_VERSION: {{ service_version }}
      DB_HOSTNAME: ...
      REDIS_HOSTNAME: ...
      DB_USERNAME: ...
      DB_PASSWORD: ...
      DB_DATABASE_NAME: ...
    networks:
      - postgres_postgresnet
      - redis_redisnet
    restart: unless-stopped

networks:
  traefik_network:
    external: true
  postgres_postgresnet:
    external: true
  redis_redisnet:
    external: true
```

### Your .env content

```Shell
n/a
```

### Reproduction steps

```bash
n/a
```

**Comment by lgo** [comment]
hmm, unfortunately after getting this unwedged with doing the upgrade in steps I realized I had no information on the `REINDEX` error which wasn't logged and likely can't reproduce. I still don't quite know what the first error was that caused me to go down this rabbit hole, but I suspect it was the crash as reported or the lack of permissions on `pg_vector_index_stat`.

**Comment by mmomjian** [comment]
Seems like it’s working now, and we already have a fix for some of this. Thanks.

**Comment by rursache** [comment]
I also had migration issues (again, really getting tired of these) and eventually ended up using @lgo's method to finally fix it!

I started with this error:
```sh
2024-07-17T17:17:35.084018789Z Starting api worker
2024-07-17T17:17:37.209751774Z [Nest] 776  - 07/17/2024, 8:17:37 PM     LOG [Api:EventRepository] Initialized websocket server
2024-07-17T17:17:37.262773773Z Migration "AddFaceSearchRelation1718486162779" failed, error: type "vector" does not exist
2024-07-17T17:17:37.265813209Z QueryFailedError: type "vector" does not exist
2024-07-17T17:17:37.265831960Z     at PostgresQueryRunner.query (/app/immich/server/node_modules/typeorm/driver/postgres/PostgresQueryRunner.js:219:19)
2024-07-17T17:17:37.265839944Z     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
2024-07-17T17:17:37.265843723Z     at async AddFaceSearchRelation1718486162779.up (/app/immich/server/dist/migrations/1718486162779-AddFaceSearchRelation.js:23:9)
2024-07-17T17:17:37.265846057Z     at async MigrationExecutor.executePendingMigrations (/app/immich/server/node_modules/typeorm/migration/MigrationExecutor.js:225:17)
2024-07-17T17:17:37.265848262Z     at async DataSource.runMigrations (/app/immich/server/node_modules/typeorm/data-source/DataSource.js:265:35)
2024-07-17T17:17:37.265850431Z     at async DatabaseRepository.runMigrations (/app/immich/server/dist/repositories/database.repository.js:169:9)
2024-07-17T17:17:37.265852585Z     at async /app/immich/server/dist/services/database.service.js:136:17
2024-07-17T17:17:37.265854680Z     at async /app/immich/server/dist/repositories/database.repository.js:177:23 {
2024-07-17T17:17:37.265856800Z   query: '\n' +
2024-07-17T17:17:37.265858992Z     '      CREATE TABLE face_search (\n' +
2024-07-17T17:17:37.265861156Z     '      "faceId"  uuid PRIMARY KEY REFERENCES asset_faces(id) ON DELETE CASCADE,\n' +
2024-07-17T17:17:37.265877573Z     '      embedding  vector(512) NOT NULL )',
2024-07-17T17:17:37.265880111Z   parameters: undefined,
2024-07-17T17:17:37.265882580Z   driverError: error: type "vector" does not exist
2024-07-17T17:17:37.265884828Z       at /app/immich/server/node_modules/pg/lib/client.js:526:17
2024-07-17T17:17:37.265887022Z       at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
2024-07-17T17:17:37.265889259Z       at async PostgresQueryRunner.query (/app/immich/server/node_modules/typeorm/driver/postgres/PostgresQueryRunner.js:184:25)
2024-07-17T17:17:37.265891527Z       at async AddFaceSearchRelation1718486162779.up (/app/immich/server/dist/migrations/1718486162779-AddFaceSearchRelation.js:23:9)
2024-07-17T17:17:37.265893820Z       at async MigrationExecutor.executePendingMigrations (/app/immich/server/node_modules/typeorm/migration/MigrationExecutor.js:225:17)
2024-07-17T17:17:37.265896088Z       at async DataSource.runMigrations (/app/immich/server/node_modules/typeorm/data-source/DataSource.js:265:35)
2024-07-17T17:17:37.265898295Z       at async DatabaseRepository.runMigrations (/app/immich/server/dist/repositories/database.repository.js:169:9)
2024-07-17T17:17:37.265900499Z       at async /app/immich/server/dist/services/database.service.js:136:17
2024-07-17T17:17:37.265903260Z       at async /app/immich/server/dist/repositories/database.repository.js:177:23 {
2024-07-17T17:17:37.265905470Z     length: 94,
2024-07-17T17:17:37.265907589Z     severity: 'ERROR',
2024-07-17T17:17:37.265909748Z     code: '42704',
2024-07-17T17:17:37.265911817Z     detail: undefined,
2024-07-17T17:17:37.265913901Z     hint: undefined,
2024-07-17T17:17:37.265915991Z     position: '131',
2024-07-17T17:17:37.265918092Z     internalPosition: undefined,
2024-07-17T17:17:37.265920167Z     internalQuery: undefined,
2024-07-17T17:17:37.265922249Z     where: undefined,
2024-07-17T17:17:37.265924314Z     schema: undefined,
2024-07-17T17:17:37.265926432Z     table: undefined,
2024-07-17T17:17:37.265928495Z     column: undefined,
2024-07-17T17:17:37.265930561Z     dataType: undefined,
2024-07-17T17:17:37.265932630Z     constraint: undefined,
2024-07-17T17:17:37.265934708Z     file: 'parse_type.c',
2024-07-17T17:17:37.265936786Z     line: '270',
2024-07-17T17:17:37.265938865Z     routine: 'typenameType'
2024-07-17T17:17:37.265940991Z   },
2024-07-17T17:17:37.265943087Z   length: 94,
2024-07-17T17:17:37.265945176Z   severity: 'ERROR',
2024-07-17T17:17:37.265947262Z   code: '42704',
2024-07-17T17:17:37.265949315Z   detail: undefined,
2024-07-17T17:17:37.265951377Z   hint: undefined,
2024-07-17T17:17:37.265953437Z   position: '131',
2024-07-17T17:17:37.265958711Z   internalPosition: undefined,
2024-07-17T17:17:37.265961584Z   internalQuery: undefined,
2024-07-17T17:17:37.265963784Z   where: undefined,
2024-07-17T17:17:37.265965919Z   schema: undefined,
2024-07-17T17:17:37.265968018Z   table: undefined,
2024-07-17T17:17:37.265970120Z   column: undefined,
2024-07-17T17:17:37.265972240Z   dataType: undefined,
2024-07-17T17:17:37.265974360Z   constraint: undefined,
2024-07-17T17:17:37.265976489Z   file: 'parse_type.c',
2024-07-17T17:17:37.265978607Z   line: '270',
2024-07-17T17:17:37.265980771Z   routine: 'typenameType'
2024-07-17T17:17:37.265982961Z }
2024-07-17T17:17:37.266859915Z node:internal/process/promises:391
2024-07-17T17:17:37.266878737Z     triggerUncaughtException(err, true /* fromPromise */);
2024-07-17T17:17:37.266883126Z     ^
2024-07-17T17:17:37.266887139Z 
2024-07-17T17:17:37.266890232Z QueryFailedError: type "vector" does not exist
2024-07-17T17:17:37.266892864Z     at PostgresQueryRunner.query (/app/immich/server/node_modules/typeorm/driver/postgres/PostgresQueryRunner.js:219:19)
2024-07-17T17:17:37.266895297Z     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
2024-07-17T17:17:37.266897608Z     at async AddFaceSearchRelation1718486162779.up (/app/immich/server/dist/migrations/1718486162779-AddFaceSearchRelation.js:23:9)
2024-07-17T17:17:37.266899989Z     at async MigrationExecutor.executePendingMigrations (/app/immich/server/node_modules/typeorm/migration/MigrationExecutor.js:225:17)
2024-07-17T17:17:37.266902337Z     at async DataSource.runMigrations (/app/immich/server/node_modules/typeorm/data-source/DataSource.js:265:35)
2024-07-17T17:17:37.266904684Z     at async DatabaseRepository.runMigrations (/app/immich/server/dist/repositories/database.repository.js:169:9)
2024-07-17T17:17:37.266906985Z     at async /app/immich/server/dist/services/database.service.js:136:17
2024-07-17T17:17:37.266909252Z     at async /app/immich/server/dist/repositories/database.repository.js:177:23 {
2024-07-17T17:17:37.266911567Z   query: '\n' +
2024-07-17T17:17:37.266913853Z     '      CREATE TABLE face_search (\n' +
2024-07-17T17:17:37.266916197Z     '      "faceId"  uuid PRIMARY KEY REFERENCES asset_faces(id) ON DELETE CASCADE,\n' +
2024-07-17T17:17:37.266918624Z     '      embedding  vector(512) NOT NULL )',
2024-07-17T17:17:37.266920883Z   parameters: undefined,
2024-07-17T17:17:37.266923600Z   driverError: error: type "vector" does not exist
2024-07-17T17:17:37.266926004Z       at /app/immich/server/node_modules/pg/lib/client.js:526:17
2024-07-17T17:17:37.266928389Z       at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
2024-07-17T17:17:37.266930745Z       at async PostgresQueryRunner.query (/app/immich/server/node_modules/typeorm/driver/postgres/PostgresQueryRunner.js:184:25)
2024-07-17T17:17:37.266943864Z       at async AddFaceSearchRelation1718486162779.up (/app/immich/server/dist/migrations/1718486162779-AddFaceSearchRelation.js:23:9)
2024-07-17T17:17:37.266946641Z       at async MigrationExecutor.executePendingMigrations (/app/immich/server/node_modules/typeorm/migration/MigrationExecutor.js:225:17)
2024-07-17T17:17:37.266949050Z       at async DataSource.runMigrations (/app/immich/server/node_modules/typeorm/data-source/DataSource.js:265:35)
2024-07-17T17:17:37.266951991Z       at async DatabaseRepository.runMigrations (/app/immich/server/dist/repositories/database.repository.js:169:9)
2024-07-17T17:17:37.266954346Z       at async /app/immich/server/dist/services/database.service.js:136:17
2024-07-17T17:17:37.266956628Z       at async /app/immich/server/dist/repositories/database.repository.js:177:23 {
2024-07-17T17:17:37.266958892Z     length: 94,
2024-07-17T17:17:37.266961117Z     severity: 'ERROR',
2024-07-17T17:17:37.266963319Z     code: '42704',
2024-07-17T17:17:37.266965505Z     detail: undefined,
2024-07-17T17:17:37.266967710Z     hint: undefined,
2024-07-17T17:17:37.266969889Z     position: '131',
2024-07-17T17:17:37.266972704Z     internalPosition: undefined,
2024-07-17T17:17:37.266975796Z     internalQuery: undefined,
2024-07-17T17:17:37.266978002Z     where: undefined,
2024-07-17T17:17:37.266980202Z     schema: undefined,
2024-07-17T17:17:37.266982372Z     table: undefined,
2024-07-17T17:17:37.266984531Z     column: undefined,
2024-07-17T17:17:37.266986704Z     dataType: undefined,
2024-07-17T17:17:37.266989174Z     constraint: undefined,
2024-07-17T17:17:37.266992752Z     file: 'parse_type.c',
2024-07-17T17:17:37.266996284Z     line: '270',
2024-07-17T17:17:37.266999134Z     routine: 'typenameType'
2024-07-17T17:17:37.267001369Z   },
2024-07-17T17:17:37.267003549Z   length: 94,
2024-07-17T17:17:37.267005713Z   severity: 'ERROR',
2024-07-17T17:17:37.267007897Z   code: '42704',
2024-07-17T17:17:37.267010056Z   detail: undefined,
2024-07-17T17:17:37.267012203Z   hint: undefined,
2024-07-17T17:17:37.267014385Z   position: '131',
2024-07-17T17:17:37.267016532Z   internalPosition: undefined,
2024-07-17T17:17:37.267018753Z   internalQuery: undefined,
2024-07-17T17:17:37.267020912Z   where: undefined,
2024-07-17T17:17:37.267023062Z   schema: undefined,
2024-07-17T17:17:37.267025226Z   table: undefined,
2024-07-17T17:17:37.267027379Z   column: undefined,
2024-07-17T17:17:37.267029541Z   dataType: undefined,
2024-07-17T17:17:37.267031721Z   constraint: undefined,
2024-07-17T17:17:37.267037274Z   file: 'parse_type.c',
2024-07-17T17:17:37.267039661Z   line: '270',
2024-07-17T17:17:37.267041881Z   routine: 'typenameType'
2024-07-17T17:17:37.267044072Z }
2024-07-17T17:17:37.267046239Z 
2024-07-17T17:17:37.267048399Z Node.js v20.15.1
2024-07-17T17:17:37.291681661Z api worker exited with code 1
```

which after some time wasted troubleshooting I ended up fixing by making the postgresql user have `superuser` permissions.
not great advice but i got tired of these problems.

moving on, I reached @lgo issue which I fixed by firstly starting up the `v1.106.4` tagged immich, then `v1.107.0` and finally reaching the latest `v1.108.x` version
